### DISCUSSION TITLE
How to load a q-img using a relative path from an object?

### DISCUSSION THREAD

**Discussion by BenJackGill** [quasarframework/quasar#13340] [General - Components / Directives / etc]
In my Vue 3 / Quasar / TypeScript app I have an object holding the relative path to my images. And I'm trying to use that to load the relative path for a `q-img`.

Here is the example code:

```TypeScript
<template>
  <q-page class="flex flex-center">
    <q-item-section
      v-for="(plan, index) in plans"
      :key="index"
      :class="plan.class"
      class="col"
    >
      <q-img :src="plan.imageSrc" style="margin: auto; max-width: 64px" />
    </q-item-section>
  </q-page>
</template>

<script setup lang="ts">
const plans = [
  {
    imageSrc: "~/src/assets/paper-plane.png",
    monthlyPrice: "699",
    class: "starter",
  },
  {
    imageSrc: "~/src/assets/airport.png",
    monthlyPrice: "2,789",
    class: "growth",
  },
];
</script>
```

It should display the image at the relative path found at `plan.imageSrc` but no images are shown.

And the console also throws these errors:

```
http://localhost:8080/~/src/assets/paper-plane.png 404 (Not Found)
http://localhost:8080/~/src/assets/airport.png 404 (Not Found)
```

Also I tried using `require` as per below, and it also does not work:

```
 <q-img :src="require(plan.imageSrc)" style="margin: auto; max-width: 64px" />
```

It throws console errors like this:

```
Error: Cannot find module '~/src/assets/paper-plane.png'
```

What am I doing wrong?

**Comment by metalsadman** [comment]
see https://quasar.dev/quasar-cli-webpack/handling-assets#vue-binding-requires-statics-only

https://quasar.dev/quasar-cli-webpack/handling-assets#getting-asset-paths-in-javascript
